src/ProxyInterfaceSourceGenerator/Extensions/NamedTypeSymbolExtensions.cs

```diff
@@ -60,4 +60,41 @@ public static string ResolveFullProxyClassName(this INamedTypeSymbol namedTypeSy
             $"{namedTypeSymbol}Proxy" :
             $"{namedTypeSymbol}Proxy<{string.Join(", ", namedTypeSymbol.TypeArguments.Select(ta => ta.Name))}>";
     }
+
+    public static List<INamedTypeSymbol> ResolveImplementedInterfaces(this INamedTypeSymbol symbol, bool proxyBaseClasses)

```

Comment:
Can you make a UnitTest for this new method?
Or create an unit-test in ProxyInterfaceSourceGeneratorTest which tests this scenario which is now fixed by this PR?

Reply:
I created a test inside ProxyInterfaceSourceGeneratorTests. 384cd65a73293146211bde9621553d0d444b8d55